backend/api/views.py

```diff
@@ -0,0 +1,399 @@
+    language_prob.extend(detect_en)
+
+    detect_fr = [str(i) for i in detect_langs(response2)]
+    language_prob.extend(detect_fr)
+
+    detect_es = [str(i) for i in detect_langs(response3)]
+    language_prob.extend(detect_es)
+
+    return list(language_prob)
+
+
+def lang_prob(lang_probability):
+    """
+    Function for seperating the probability and language codes
+    """
+    language_code = []
+
+    # Seperate probs and language codes into 2 lists
+    for i in range(0, len(lang_probability)):
+        code = lang_probability[i][0:2]
+        language_code.append(code)
+        lang_probability[i] = lang_probability[i][3:]
+
+    # Flatten list
+    probs = [float(i) for i in lang_probability]
+
+    return language_code, probs
+
+
+def ISO_639(langauge_code, probability):
+    """
+    Function to get the language ISO
+    """
+    lang_counter = Counter(langauge_code)
+
+    test_val = list(lang_counter.values())[0]
+    res = True
+    
+    for i in lang_counter:
+        if lang_counter[i] != test_val:
+            res = False
+            break
+
+    if res == True:
+        d = defaultdict(list)
+        for key, value in zip(langauge_code, probability):
+            d[key].append(value)
+
+        mean_probs = {key: np.mean(val) for key, val in dict(d).items()}
+
+        ISO_639_2 = str(max(mean_probs, key = mean_probs.get))
+
+    else:
+        ISO_639_2 = max(lang_counter, key = lang_counter.get)
+    
+    return ISO_639_2
+
+
+def exit_input(message):
+    if message.lower() == "exit please":
+        return 1
+    return 0
+
+
+def speak_response(lang_ISO, res_en2lang, lang_voice):
+    """
+    Translate audio response into different languages
+    """
+    if lang_ISO in lang_voice:
+        engine = pyttsx3.init()
+        if lang_ISO == "es":
+            engine.setProperty("rate", 150)  # Speed of speech (words per minute)
+        voices = engine.getProperty("voices")
+        engine.setProperty("voice", voices[lang_voice[lang_ISO]].id)
+        engine.say(res_en2lang)
+        engine.runAndWait()
+
+    # else:
+    #     engine = pyttsx3.init()
+    #     voices = engine.getProperty("voices")
+    #     engine.setProperty("voice", voices[lang_voice[1]].id)
+    #     engine.say("Language not in the system. Please try again in English, French or Spnaish.")
+    #     engine.runAndWait()
+
+
+@api_view(["GET"])
+def main(request):
+
+    lang_voice = {"en": 1, "es": 2, "fr": 3}
+
+    intents = load_intents()
+
+    try:
+        English, French, Spanish = capture_and_recognize(request)
+
+    except:
+        details = {"User_Request": " ",
+                   "Chatbot_Response": "Invalid audio input given. Please press the button to speak to BabelBot!"}
+        
+        lang_ISO = "en"
+        res_en2lang = "Invalid audio input given. Please press the button to speak to BabelBot!"
+        speak_response(lang_ISO, res_en2lang, lang_voice)
+
+        return render(request,"kiosk.html", details)
```

Comment:
What's the difference between "invalid audio" here, and "no audio" in `capture_and_recognize`? Do we need multiple try except blocks, or can we move this try except block into the function?

Reply:
The idea was that 'invalid audio' would be for audio it can't decipher and 'no audio' would be for when no audio was fed. I can play around with this and see if I can just make it into one instead.